Sources/TuistGenerator/Generator/ConfigGenerator.swift

```diff
@@ -66,23 +64,29 @@ final class ConfigGenerator: ConfigGenerating {
         pbxproj.add(object: configurationList)
         pbxTarget.buildConfigurationList = configurationList
 
-        try generateTargetSettingsFor(target: target,
-                                      buildConfiguration: .debug,
-                                      configuration: target.settings?.debug,
-                                      fileElements: fileElements,
-                                      graph: graph,
-                                      pbxproj: pbxproj,
-                                      configurationList: configurationList,
-                                      sourceRootPath: sourceRootPath)
-
-        try generateTargetSettingsFor(target: target,
-                                      buildConfiguration: .release,
-                                      configuration: target.settings?.release,
-                                      fileElements: fileElements,
-                                      graph: graph,
-                                      pbxproj: pbxproj,
-                                      configurationList: configurationList,
-                                      sourceRootPath: sourceRootPath)
+        let projectBuildConfigurations = projectSettings.configurations.keys
+        let targetConfigurations = target.settings?.configurations ?? [:]
+        let targetBuildConfigurations = targetConfigurations.keys
+        let buildConfigurations = Set(projectBuildConfigurations).union(targetBuildConfigurations)
+        let configurationsTuples: [(BuildConfiguration, Configuration?)] = buildConfigurations.map {
+            if let configuration = target.settings?.configurations[$0] {
```

Comment:
Nitpick: It's hard to guess what `$0` refers to here. What about naming the closure argument?